src/contracts/BCoWPool.sol

```diff
@@ -0,0 +1,227 @@
+      value := tload(COMMITMENT_SLOT)
+    }
+  }
+
+  /**
+   * @notice This function checks that the input order is admissible for the
+   * constant-product curve for the given trading parameters.
+   * @param order `GPv2Order.Data` of a discrete order to be verified.
+   */
+  function verify(GPv2Order.Data memory order) public view {
+    Record memory inRecord = _records[address(order.sellToken)];
+    Record memory outRecord = _records[address(order.buyToken)];
+
+    require(inRecord.bound, 'BCoWPool: TOKEN_NOT_BOUND');
+    require(outRecord.bound, 'BCoWPool: TOKEN_NOT_BOUND');
+
+    require(order.validTo < block.timestamp + MAX_ORDER_DURATION, 'BCoWPool: ORDER_EXPIRED');
+    require(order.feeAmount == 0, 'BCoWPool: FEE_NOT_ZERO');
+    require(order.kind == GPv2Order.KIND_SELL, 'BCoWPool: INVALID_OPERATION');
+    require(order.buyTokenBalance == GPv2Order.BALANCE_ERC20, 'BCoWPool: INVALID_BALANCE');
+    require(order.sellTokenBalance == GPv2Order.BALANCE_ERC20, 'BCoWPool: INVALID_BALANCE');
+
+    uint256 tokenAmountOut = calcOutGivenIn({
+      tokenBalanceIn: order.sellToken.balanceOf(address(this)),
+      tokenWeightIn: inRecord.denorm,
+      tokenBalanceOut: order.buyToken.balanceOf(address(this)),
+      tokenWeightOut: outRecord.denorm,
+      tokenAmountIn: order.sellAmount,
+      swapFee: 0
+    });
+
+    require(tokenAmountOut >= order.buyAmount, 'BCoWPool: INSUFFICIENT_OUTPUT_AMOUNT');
+  }
+
+  /**
+   * @dev Computes an identifier that uniquely represents the parameters in
+   * the function input parameters.
+   * @param appData Bytestring that decodes to `AppData`
+   * @return The hash of the input parameter, intended to be used as a unique identifier
+   */
+  function hash(bytes32 appData) public pure returns (bytes32) {
+    return keccak256(abi.encode(appData));
+  }
+
+  /**
+   * @notice Approves the spender to transfer an unlimited amount of tokens
+   * and reverts if the approval was unsuccessful.
+   * @param token The ERC-20 token to approve.
+   * @param spender The address that can transfer on behalf of this contract.
+   */
+  function _approveUnlimited(IERC20 token, address spender) internal {
+    token.approve(spender, type(uint256).max);
+  }
+
+  /// @inheritdoc BPool
+  /// @dev Grants infinite approval to the vault relayer for all tokens in the
+  /// pool after the finalization of the setup.
+  function finalize() public override {
+    super.finalize();
+    for (uint256 i; i < _tokens.length; i++) {
+      _approveUnlimited(IERC20(_tokens[i]), VAULT_RELAYER);
+    }
+  }
+
+  /**
+   * @notice This function triggers a revert if either (1) the order hash does
+   * not match the current commitment, or (2) in the case of a commitment to
+   * `EMPTY_COMMITMENT`, the non-constant parameters of the order from
+   * `getTradeableOrder` don't match those of the input order.
+   * @param orderHash the hash of the current order as defined by the
+   * `GPv2Order` library.
+   */
```

Comment:
i think this description is outdated since we don't have a `getTradeableOrder` flow.
this method should fail if:
1) `orderHash` doesn't match `commitment()`

let's remove the `EMPTY_COMMITMENT` exception here.

Reply:
i would remove this whole method, and replace it with a 1-liner within `isValidSignature`:
```solidity
if (orderHash != commitment()) revert OrderDoesNotMatchCommitmentHash();
```